### PR TITLE
Improve error handling & reporting.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request makes a few improvements to how we report errors to users:

⚠️ Makes sure we report string errors to New Relic. The [`newrelic.noticeError`](https://docs.newrelic.com/docs/browser/new-relic-browser/browser-agent-spa-api/notice-error) API expects an `Error`, and sometimes we'll pass a string (say from a GraphQL issue). This ensures those non-traditional errors appear in Error Analytics too. ad67c39

🔂 Makes sure we don't report errors multiple times if the app re-renders. aec837f

ℹ️ Includes technical details in user-facing errors (so that they're visible in screenshots we get in Zendesk tickets). Bonus: the "support squad" link auto-fills on the linked helpdesk form! 3ad35a6

🔲 Adds a root-level error boundary, so any uncaught errors will show our ErrorPage. [e0173cd](https://github.com/DoSomething/phoenix-next/commit/e0173cdada439a66d204908683f1cba11814047e?w=1)

### How should this be reviewed?

It's probably easiest to review commit-by-commit for context.

### Any background context you want to provide?

Here's how the root-level `ErrorPage` (for uncaught errors) now looks. This will allow us to see the error message, first line of the stack trace, timestamp (for correlating to server logs), and user ID (again, for correlating to server-side logs if they're logged in):

Screen Shot 2020-04-27 at 2 27 34 PM

And the updated ErrorBlock, for individual component errors:

Screen Shot 2020-04-27 at 2 27 55 PM

In both cases, these will pre-fill the ticket body on the "New Request" page using this snippet that I added into the Zendesk template (see `templates/new_request_page.hbs` on [Zendesk](https://help.dosomething.org/theming/workbench)):

```js
<script type="text/javascript">
$(document).ready(function() {
  // Don't bother autofilling for old browsers.
  if (!window.URLSearchParams) {
    return;
  }

  const urlParams = new URLSearchParams(window.location.search);
  const technicalDetails = urlParams.get('technicalDetails');
  if (technicalDetails) {
    $('#request_description').val('\n\n--------\nTechnical Details:\n' + technicalDetails);
  }
});
</script>
```

### Relevant tickets

References [Pivotal #172384322](https://www.pivotaltracker.com/story/show/172384322).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [x] Added screenshots of front-end changes on ~small, medium, and~ large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/utilities/ErrorDetails.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const ErrorDetails = ({ error }) => {
+  const context = `${+new Date()}, ${window.AUTH ? window.AUTH.id : 'N/A'}`;
+
+  let message = error;
+  if (error instanceof Error) {
+    const trace = error.stack.toString().split(/\r\n|\n/)[0];
+    message = `${error.message} ${trace}, ${context}`;
+  }
+
+  const technicalDetails = `${message}, ${context}`;
```
Comment (mendelB) [inline resources/assets/components/utilities/ErrorDetails.js:16]: Why do we add the `context` both to the `message` and the `technicalDetails`?

File: resources/assets/components/utilities/ErrorDetails.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <>
+      <p className="text-sm text-gray-800 text-center m-0">
+        If you continue to run into problems, contact our{' '}
+        <a
+          href={makeUrl(HELP_REQUEST_LINK, { technicalDetails })}
```
Comment (mendelB) [inline resources/assets/components/utilities/ErrorDetails.js:23]: 🔥 I did not know this was a thing!

File: resources/assets/components/utilities/ErrorDetails.js (hunk 1 above)
Comment (mendelB) [reply resources/assets/components/utilities/ErrorDetails.js:16]: Also, would it be helpful to prefix with like `ID:`? I found the log slightly hard to decipher with my poor 👀...though that could also just be the nature of logs!

Comment (DFurnes) [reply resources/assets/components/utilities/ErrorDetails.js:16]: Oh that's 100% an accident! And yeah, that's a good idea!

File: resources/assets/components/blocks/ErrorBlock/ErrorBlock.js [hunk 3]
```diff
@@ -1,32 +1,24 @@
...
-const DEBUGGING = process.env.NODE_ENV !== 'production';
+import ErrorDetails from '../../utilities/ErrorDetails';
 
 const ErrorBlock = ({ error }) => {
   // Print error to console & report to New Relic:
-  if (error) {
-    console.error(`[ErrorBlock] ${error}`);
-    report(error);
-  }
+  useEffect(() => report(error), []);
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/blocks/ErrorBlock/ErrorBlock.js:11]: Oooh maybe! Certainly shouldn't hurt!

File: resources/assets/components/utilities/ErrorDetails.js (hunk 2 above)
Comment (DFurnes) [reply resources/assets/components/utilities/ErrorDetails.js:23]: It wasn't! I added it in with a jQuery snippet on our Zendesk template. It is, somewhat shockingly IMHO, not a built-in feature (despite being a long-standing [feature request](https://support.zendesk.com/hc/pt-br/community/posts/203439446-Pre-populate-ticket-submission-form-field-via-query-string-Help-Center-?page=3#comments)).

File: resources/assets/components/utilities/ErrorDetails.js (hunk 1 above)
Comment (DFurnes) [reply resources/assets/components/utilities/ErrorDetails.js:16]: Made those two updates in fd5a92b.

File: resources/assets/components/blocks/ErrorBlock/ErrorBlock.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/blocks/ErrorBlock/ErrorBlock.js]: Updated in 0063616.

File: resources/assets/components/utilities/ErrorDetails.js (hunk 2 above)
Comment (mendelB) [reply resources/assets/components/utilities/ErrorDetails.js:23]: Oh neato!! (I misread that final paragraph. agreed, that feels like a no brainer).

**Pull request merged**